rmf_traffic/src/rmf_traffic/Conflict.cpp

```diff
@@ -121,54 +121,116 @@ std::vector<ConflictData> DetectConflict::between(
 }
 
 //==============================================================================
-bool DetectConflict::broad_phase(
-    const Trajectory& trajectory_a,
-    const Trajectory& trajectory_b)
+
+namespace {
+
+using BoundingBox = std::pair<Eigen::Vector2d, Eigen::Vector2d>;
+
+double evaluate_spline(
+    const Eigen::Vector4d& coeffs,
+    const double t)
 {
-  std::size_t min_size = std::min(trajectory_a.size(), trajectory_b.size());
-  if(min_size < 2)
+  // Assume time is parameterized [0,1]
+  return (coeffs[3] * t * t * t
+      + coeffs[2] * t * t
+      + coeffs[1] * t
+      + coeffs[0]);
+}
+
+void get_local_extrema(
+    const Eigen::Vector4d& coeffs,
+    std::array<double, 2>& extrema)
+{
+  std::vector<double> t_sols;
+  // Store boundary values as potential extrema
+  t_sols.emplace_back(evaluate_spline(coeffs, 0));
+  t_sols.emplace_back(evaluate_spline(coeffs, 1));
+
+  // When derivate of spline motion is not quadratic
+  if (coeffs[3] == 0)
   {
-    throw invalid_trajectory_error::Implementation
-        ::make_segment_num_error(min_size);
+    if (coeffs[2] != 0)
+      t_sols.emplace_back(-coeffs[1] / coeffs[2]);
   }
+  else
+  {
+    // Calculate the discriminant otherwise
+    double D = (4 * coeffs[2] * coeffs[2]) - 12 * coeffs[3] * coeffs[1];
 
-  if(trajectory_a.get_map_name() != trajectory_b.get_map_name())
-    return false;
-
-  const auto* t_a0 = trajectory_a.start_time();
-  const auto* t_bf = trajectory_b.finish_time();
+    if (D < 0)
+    {
+      // Do nothing
+    }
+    else if (D == 0)
+    {
+      double t = (-2 * coeffs[2]) / (6 * coeffs[3]);
+      double extrema = evaluate_spline(coeffs, t);
+      t_sols.emplace_back(extrema);
+    }
+    else
+    {
+      double t1 = ((-2 * coeffs[2]) + std::sqrt(D)) / (6 * coeffs[3]);
+      double t2 = ((-2 * coeffs[2]) - std::sqrt(D)) / (6 * coeffs[3]);
 
-  // Neither of these can be null, because both trajectories should have at
-  // least two elements.
-  assert(t_a0 != nullptr);
-  assert(t_bf != nullptr);
+      double extrema1 = evaluate_spline(coeffs, t1);
+      double extrema2 = evaluate_spline(coeffs, t2);
 
-  if(*t_bf < *t_a0)
-  {
-    // If Trajectory `b` finishes before Trajectory `a` starts, then there
-    // cannot be any conflict.
-    return false;
+      t_sols.emplace_back(extrema1);
+      t_sols.emplace_back(extrema2);
+    }
   }
 
-  const auto* t_b0 = trajectory_b.start_time();
-  const auto* t_af = trajectory_a.finish_time();
+  extrema[0] = *std::min_element(t_sols.begin(), t_sols.end());
```

Comment:
I recommend adding `assert(!t_sols.empty());` before this line just as a precaution.

Reply:
Noted.